apps/settings/components/form_card/index.coffee

```diff
@@ -12,15 +12,22 @@ module.exports = ($el) ->
 
     serializer = new Serializer $el
 
+    onSubmit = if onSubmit then onSubmit else ->
+      promise = new Promise
+      promise.resolve()
+
     $submit
       .prop 'disabled', true
       .text 'Saving...'
 
-    Promise $.ajax
-      url: $el.data('action')
-      method: $el.data('method')
-      data: serializer.data()
-
+    Promise.all [
+      $.ajax(
+        url: $el.data('action')
+        method: $el.data('method')
+        data: serializer.data()
+      ),
+      onSubmit()
```

Comment:
What is this onSubmit?

Reply:
Oh you're trying to optionally include a promise returning fn.

Reply:
Two things: this is fine but very awkward, that part doesn't matter much. But $.ajax doesn't return an actual promise. You need to wrap it in Promise